app/views/viewlet-manager.js

```diff
@@ -316,12 +326,39 @@ YUI.add('juju-viewlet-manager', function(Y) {
             // parameter needed to be added so that multiple views could
             // be rendered into a single container
             if (!viewToCheck.slot) {
-              viewToCheck.hide();
+              // If this is a React component then it doesn't have a hide
+              if (typeof viewToCheck.hide === 'function') {
+                viewToCheck.hide();
+              } else if (viewToCheck.renderedComponent) {
+                // It's a React component but it may not be rendered so we need
+                // to check if it has a DOM node.
+                var domNode;
+                try {
+                  domNode = React.findDOMNode(viewToCheck.renderedComponent);
+                } catch (e) {
+                  // do nothing if it's not in the DOM.
+                }
+                if (domNode) {
+                  React.unmountComponentAtNode(domNode.parentNode);
```

Comment:
Y.ReactMgr.destroy() could even give a nice Y.View api so that the code can be kept the same perhaps. show/hide on the container div or the like.